### ISSUE TITLE
IndentationCheck: no violation on not indented method header inside anonymous class

### ISSUE THREAD

**Issue by pirat9600q** [checkstyle/checkstyle#618]
Input.java
```java
public class Input {
  void f() {
    Object b = new Thread(new Runnable() {
      @Override
public void run() { //indent:0, expected:6; no violation generated
        System.out.print("h");
      }
    });
  }
}
```
config.xml
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">

  <property name="severity" value="error"/>

  <property name="fileExtensions" value="java, properties, xml"/>

  <module name="TreeWalker">
    <property name="tabWidth" value="4"/>

    <module name="Indentation">
      <property name="basicOffset" value="2"/>
      <property name="braceAdjustment" value="0"/>
      <property name="lineWrappingIndentation" value="4"/>
    </module>
  </module>
</module>
```
Invocation:
```
java -cp checkstyle-6.3-all.jar com.puppycrawl.tools.checkstyle.Main -c config.xml Input.java
```

actual output:
```
Starting audit...
Audit done.
```
expected violation on line 5

**Comment by robertwhitebit** [comment]
This problem is far more critical because it effects all methods (not only anonymous classes ones) which are using the @Override annotation or any other annotation.

Using your configuration and checkstyle version 6.9, following simple class does *not* produce a warning as well.
```java
public class Input implements Comparable<Input> {
  @Override
public int compareTo(final Input otherInput) {
    return 0;
  }

  @Ignore
      public void h() {
  }
}
```

**Comment by romani** [comment]
Almost completed but abandoned update https://github.com/checkstyle/checkstyle/pull/9195#issuecomment-1836963836 , everyone welcome to reuse this code to finish a fix
